Doc/library/annotationlib.rst

```diff
@@ -0,0 +1,183 @@
+:mod:`!annotationlib` --- Functionality for introspecting annotations
+=====================================================================
+
+.. module:: annotationlib
+   :synopsis: Functionality for introspecting annotations
+
+
+**Source code:** :source:`Lib/annotationlib.py`
+
+.. testsetup:: default
+
+   import annotationlib
+   from annotationlib import *
+
+--------------
+
+The :mod:`!annotationlib` module provides tools for introspecting :term:`annotations <annotation>`
+on modules, classes, and functions.
+
+.. function:: call_annotate_function(annotate, format, *, owner=None)
+
+   Call the :term:`annotate function` *annotate* with the given *format*, a member of the :class:`Format`
+   enum, and return the annotations dictionary produced by the function.
+
+   Annotate functions generated by the compiler for functions, classes, and modules support only
+   the :attr:`~Format.VALUE` format when called directly. This function calls the annotate function
+   in a special environment that allows it to produce annotations in the other formats.
```

Comment:
I think this might be worth emphasising, as a reader might be confused as to why this indirection is needed.

Perhaps (with words stolen from 749):

```suggestion
   This helper function is required because annotate functions generated by
   the :term:`bytecode` compiler for functions, classes, and modules only support
   the :attr:`~Format.VALUE` format when called directly.
   To support other :class:`Format` values, this function calls the annotate function
   in a special :PEP:`"fake globals" <649#the-stringizer-and-the-fake-globals-environment>`
   environment that allows it to produce annotations in the other formats.
   This is a useful building block when implementing functionality
   that needs to partially evaluate annotations while a class is being constructed.
```

Reply:
Thanks, pushing a version of this.